File: programmers/javascript/level1/소수찾기.js
const n = 10;

function solution(n) {
  let arr = [0, 0];
  var answer = 0;
  // 입력받은 수 만큼 배열에 모두 초기화 해둔다
  for (let i = 2; i <= n; i++) {
    arr[i] = i;
  }
  for (let i = 2; i <= Math.sqrt(n); i++) {
    if (arr[i] == 0)
      // 이미 체크된 수의 배수는 확인하지 않는다
      continue;
    for (let j = i + i; j <= n; j += i) {
      // i를 제외한 i의 배수들은 0으로 체크
      arr[j] = 0;
    }
  }
  for (let i = 1; i <= n; i++) {
    if (arr[i] !== 0) {
      answer++;
    }
  }
  return answer;
}

console.log(solution(n));

File: python/programmers/level2/소수찾기.py
from itertools import permutations


def isPrime(num):
    if num == 0 or num == 1:
        return False

    for i in range(2, num // 2 + 1):
        if (num / i) == (num // i):
            return False
    return True


def solution(numbers):
    answer = 0
    arr = list(numbers)
    newSet = set()
    for i in range(1, len(arr) + 1):
        permute = [*permutations(arr, i)]
        # print(permute)
        for x in list(permute):
            # print(x)
            newSet.add(int(''.join(map(str, x))))

    # print(newSet)
    for item in newSet:
        if isPrime(item):
            answer += 1

    return answer
